providers/adfs.go

```diff
@@ -65,36 +72,48 @@ func (p *ADFSProvider) GetLoginURL(redirectURI, state, nonce string) string {
 		extraParams.Add("nonce", nonce)
 	}
 	loginURL := makeLoginURL(p.Data(), redirectURI, url.QueryEscape(state), extraParams)
-	if p.SkipScope {
+	if p.skipScope {
 		q := loginURL.Query()
 		q.Del("scope")
 		loginURL.RawQuery = q.Encode()
 	}
 	return loginURL.String()
 }
 
-// EnrichSession to add email
+// EnrichSession calls the OIDC ProfileURL to backfill any fields missing
+// from the claims. If Email is missing, falls back to ADFS `upn` claim.
 func (p *ADFSProvider) EnrichSession(ctx context.Context, s *sessions.SessionState) error {
-	if s.Email != "" {
-		return nil
+	err := p.oidcEnrichFunc(ctx, s)
+	if err != nil || s.Email == "" {
+		// OIDC only errors if email is missing
```

Comment:
Are we sure about this? Are we sure this is what we want, should we move away from this so that the providers don't actually return errors when there is a claim missing and defer the session validation like this until our authz layer (which doesn't exist just yet)

Reply:
Interesting idea 🤔 

Even if `""` email got through, email validator would block it, except for the case where a user set `*` as email-domains.

The one area we lose is unverified emails getting rejected with an error -- that might be a downside of removing these from erroring.

Reply:
For ADFS, we could just remove the parent OIDCProvider call to the userinfo URL. I can check the original implementation and see if it even used it.